Sources/SourceKitLSPAPI/PluginTargetBuildDescription.swift

```diff
@@ -16,18 +16,20 @@ import struct PackageGraph.ResolvedModule
 
 private import class PackageLoading.ManifestLoader
 internal import struct PackageModel.ToolsVersion
-private import class PackageModel.UserToolchain
+internal import class PackageModel.UserToolchain
 import enum PackageGraph.BuildTriple
 
 struct PluginTargetBuildDescription: BuildTarget {
     private let target: ResolvedModule
     private let toolsVersion: ToolsVersion
+    private let toolchain: UserToolchain?
     let isPartOfRootPackage: Bool
 
-    init(target: ResolvedModule, toolsVersion: ToolsVersion, isPartOfRootPackage: Bool) {
+    init(target: ResolvedModule, toolsVersion: ToolsVersion, toolchain: UserToolchain?, isPartOfRootPackage: Bool) {
```

Comment:
How about we pass `BuildParameters` here which should allow us make `toolchain` field non-Optional, we can assign it this way in the initializer:

```swift
self.toolchain = if let toolchain = buildParameters.toolchain as? UserToolchain {
  toolchain
} else {
  UserToolchain(swiftSDK: .hostSwiftSDK()))
}
```

Reply:
Ended up adding a static method to get the arguments and then adding `swiftPMLibrariesLocation` to `Toolchain`, which then gets rid of this optional problem entirely.